### PR TITLE
Fix some issues

### LINKED ISSUE #346: Playground problem with serialisation of logs
The serialisation of the logs is wrong.

- showing `null` in place of the expected `undefined`:
  `console.log([undefined]); // ==>  [LOG]: [ null ]`
- not showing existing keys:
  `console.log({foo: undefined}); // ==> [LOG]: {}`

### LINKED ISSUE #1663: Docs: Classes / Constructors / Reading recommendations hidden by codeblocks (handbook v2)
**Page URL:** https://www.typescriptlang.org/docs/handbook/2/classes.html#constructors

**Issue:**

Capture d’écran 2021-03-11 à 17 29 36

### PR DESCRIPTION
**Pull request by hahmadia**
Fixes https://github.com/microsoft/TypeScript-Website/issues/346

image

Fixes https://github.com/microsoft/TypeScript-Website/issues/1663

image

### CODE REVIEW

File: packages/playground/src/sidebar/runtime.ts [hunk 1]
```diff
@@ -194,7 +194,9 @@ function rewireLoggingToElement(
       // No one needs to know an obj is an obj
       const nameWithoutObject = name && name === "Object" ? "" : name
       const prefix = nameWithoutObject ? `${nameWithoutObject}: ` : ""
-      textRep = prefix + JSON.stringify(arg, null, 2)
+
+      // JSON.stringify omits and keys with a value of undefined so to get around that, we replace undefined with __undefined__ and then do a global replace using regex with normal undefined
```
Comment (hahmadia) [inline packages/playground/src/sidebar/runtime.ts]:
```suggestion
      // JSON.stringify omits any keys with a value of undefined. To get around this, we replace undefined with the text __undefined__ and then do a global replace using regex back to keyword undefined
```

**Comment by hahmadia** [comment]
@orta Ready for review

File: packages/playground/src/sidebar/runtime.ts [hunk 2]
```diff
@@ -194,7 +194,9 @@ function rewireLoggingToElement(
       // No one needs to know an obj is an obj
       const nameWithoutObject = name && name === "Object" ? "" : name
       const prefix = nameWithoutObject ? `${nameWithoutObject}: ` : ""
-      textRep = prefix + JSON.stringify(arg, null, 2)
+
+      // JSON.stringify omits any keys with a value of undefined. To get around this, we replace undefined with the text __undefined__ and then do a global replace using regex back to keyword undefined
+      textRep = prefix + JSON.stringify(arg, (_, value) => value === undefined ? '__undefined__' : value, 2).replace(/"__undefined__"/g, 'undefined')
```
Comment (orta) [inline packages/playground/src/sidebar/runtime.ts:199]: Nice work

**Review by orta — CHANGES_REQUESTED** [review]
The undefined is great

I'm not a fan on the  block quotes though - I don't want to add _more_ HTML to the markdown docs, (having _any_ is already a bad taste but I've not had time to build a plugin/macros yet) 

So this needs to be solved without additional elements in the markdown

**Comment by hahmadia** [comment]
Hey @orta,

Here is how it looks. I've done my best to only use CSS to fix the problem without using extra HTML.

image

**Comment by orta** [comment]
Perfect, that works!

**Pull request merged**